handler/src/main/java/io/netty/handler/ssl/OpenSslKeyMaterialProvider.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2018 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.ssl;
+
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.internal.tcnative.SSL;
+
+import javax.net.ssl.X509KeyManager;
+import java.security.PrivateKey;
+import java.security.cert.X509Certificate;
+
+import static io.netty.handler.ssl.ReferenceCountedOpenSslContext.toBIO;
+
+/**
+ * Provides {@link OpenSslKeyMaterial} for a given alias.
+ */
+class OpenSslKeyMaterialProvider {
+
+    private final X509KeyManager keyManager;
+    private final String password;
+
+    OpenSslKeyMaterialProvider(X509KeyManager keyManager, String password) {
+        this.keyManager = keyManager;
```

Comment:
check not null?

Reply:
I did not check null as this is only created by internal classes which does never provide a null. I can add an assert if you like tho... Let me know

Reply:
If its all internal and it is known to be not null through other means I'm OK with omitting the check.